# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: 997e9ca2e1e9
# Full Hash: 997e9ca2e1e9d636f8f956151db81ca2a8a60c82
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r ebeb65b42212 -r 997e9ca2e1e9 dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Wed Jan 18 13:46:29 2023 +0000
+++ b/dom/workers/WorkerScope.cpp	Wed Jan 18 13:46:29 2023 +0000
@@ -192,6 +192,7 @@
                                                   DOMEventTargetHelper)
   tmp->AssertIsOnWorkerThread();
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mConsole)
+  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mModuleLoader)
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mSerialEventTarget)
   tmp->TraverseObjectsInGlobal(cb);
   // If we already exited WorkerThreadPrimaryRunnable, we will find it
@@ -206,6 +207,7 @@
                                                 DOMEventTargetHelper)
   tmp->AssertIsOnWorkerThread();
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mConsole)
+  NS_IMPL_CYCLE_COLLECTION_UNLINK(mModuleLoader)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mSerialEventTarget)
   tmp->UnlinkObjectsInGlobal();
   // If we already exited WorkerThreadPrimaryRunnable, we will find it